.github/workflows/format-check.yml

```diff
@@ -0,0 +1,25 @@
+name: Check Format
+
+on:
+  push:
+    branches: [ 'main' ]
+  pull_request:
+    branches: [ 'main' ]
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
```

Comment:
just a minor suggestion,
Can you have distribution forst and java-version next ?

This is only for readability!

Reply:
Should be possible I'll do the change